addons/web/static/tests/legacy/fields/basic_fields_tests.js

```diff
@@ -1318,6 +1318,23 @@ QUnit.module('basic_fields', {
 
         var charField = _.find(form.renderer.allFieldWidgets)[0];
         assert.strictEqual(charField.isValid(), true);
+
+        this.data.partner.fields.foo.pattern = ['Only Belgian keyboard', 'azerty'];
```

Comment:
this part deserves its own test (we prefer a lot of small tests focused on a specific feature or aspect, than a few large tests that are harder to debug)